dbt/adapters/snowflake/relation.py

```diff
@@ -106,3 +106,17 @@ def dynamic_table_config_changeset(
         if config_change_collection.has_changes:
             return config_change_collection
         return None
+
+    def as_case_sensitive(self) -> "SnowflakeRelation":
+        path_part_map = {}
+
+        for path in ComponentName:
+            if self.include_policy.get_part(path):
+                part = self.path.get_part(path)
+                if part:
+                    if self.quote_policy.get_part(path):
```

Comment:
Since it looks like Snowflake stores all of the metadata as `upper()` unless quoted, there's a part of me that wants to do this translation on the way in (e.g. inside `self.create()`) instead of each time it's needed. I think we could simplify logic elsewhere if we did that (e.g. `self._make_match_kwargs()`), and ultimately avoid needing this method.

Reply:
Created a follow-up issue to investigate: https://github.com/dbt-labs/dbt-snowflake/issues/1111

Reply:
Thanks for creating the follow up issue. Per our conversation, this suggestion is out of scope for this PR and I'm comfortable moving forward with this as-is.